RFCs/FS-1142-Extended-numeric-literal.md

```diff
@@ -0,0 +1,137 @@
+# F# RFC FS-1142 - Extended numeric literal
+
+The design suggestion [Underscores in numeric literals after prefix and before suffix](https://github.com/fsharp/fslang-suggestions/issues/718), [Extend custom numeric types to support floating point literals](https://github.com/fsharp/fslang-suggestions/issues/445) and [Hexadecimal, octal and binary custom numeric literals](https://github.com/fsharp/fslang-suggestions/issues/754) has been marked "approved in principle".
+
+This RFC covers the detailed proposal for this suggestion.
+
+- [x] Suggestion: [Underscores in numeric literals after prefix and before suffix](https://github.com/fsharp/fslang-suggestions/issues/718), [Extend custom numeric types to support floating point literals](https://github.com/fsharp/fslang-suggestions/issues/445) and [Hexadecimal, octal and binary custom numeric literals](https://github.com/fsharp/fslang-suggestions/issues/754)
+- [x] Approved in principle
+- [ ] [Implementation](https://github.com/dotnet/fsharp/pull/17242)
+- [ ] Design Review Meeting(s) with @dsyme and others invitees
+- [Discussion](https://github.com/fsharp/fslang-design/discussions/769)
+
+# Summary
+
+This RFC will allow the following things:
+
+- Underscores in numeric literals after prefix and before suffix like `0x_1` or `1_l` or mixed them up like `0x_1_l`.
+- Hexadecimal, octal, binary and floating point custom numeric literals like `0x1I` or `1.0G`
+
+# Motivation
+
+Make the language more consistent and easier to read. Enhance the custom numeric literals feature by supporting not only integers but also floats.
+
+# Detailed design
+
+1. Underscores can now be placed after prefix (`0x`, `0o`, `0b`) and before suffix (eg. `UL`, `y`, `m`, `f`). These are forbidden [previously](https://github.com/fsharp/fslang-design/blob/main/FSharp-4.1/FS-1005-underscores-in-numeric-literals.md#detailed-design).
+
+    ```fsharp
+    let pi1 = 3_.1415F      // Invalid cannot put underscores adjacent to a decimal point
+    let pi2 = 3._1415F      // Invalid cannot put underscores adjacent to a decimal point
+    let socialSecurityNumber1 = 999_99_9999_L         // OK (Invalid *previously*)
```

Comment:
I'd suggest (it is already clear that we are showing examples that are specific to this RFC anyway)
```suggestion
    let socialSecurityNumber1 = 999_99_9999_L         // OK
```